integration-test-groups/aws2/aws2-lambda/src/test/java/org/apache/camel/quarkus/component/aws2/lambda/it/Aws2LambdaTest.java

```diff
@@ -42,43 +45,50 @@
-                .pollDelay(6, TimeUnit.SECONDS) // never succeeded earlier than 6 seconds after creating the role
+        // The required role to create a function is not created immediately, so we need to retry
+        await().pollDelay(6, TimeUnit.SECONDS) // never succeeded earlier than 6 seconds after creating the role
                 .pollInterval(1, TimeUnit.SECONDS)
                 .atMost(120, TimeUnit.SECONDS)
-                .until(
-                        () -> {
-                            ExtractableResponse<?> response = RestAssured.given()
-                                    .contentType("application/zip")
-                                    .body(createInitialLambdaFunctionZip())
-                                    .post("/aws2-lambda/function/create/" + functionName)
-                                    .then()
-                                    .extract();
-                            switch (response.statusCode()) {
-                            case 201:
-                                LOG.infof("Lambda function %s created", functionName);
-                                return true;
-                            case 400:
-                                LOG.infof("Could not create Lambda function %s yet (will retry): %d %s", functionName,
-                                        response.statusCode(),
-                                        response.body().asString());
-                                return false;
-                            default:
-                                throw new RuntimeException(
-                                        "Unexpected status from /aws2-lambda/function/create " + response.statusCode() + " "
-                                                + response.body().asString());
-                            }
-                        });
+                .until(() -> {
+                    ExtractableResponse<?> response = RestAssured.given()
+                            .contentType("application/zip")
+                            .body(createInitialLambdaFunctionZip())
+                            .post("/aws2-lambda/function/create/" + functionName)
+                            .then()
+                            .extract();
+                    switch (response.statusCode()) {
+                    case 201:
+                        LOG.infof("Lambda function %s created", functionName);
+                        return true;
+                    case 400:
+                        LOG.infof("Could not create Lambda function %s yet (will retry): %d %s", functionName,
+                                response.statusCode(), response.body().asString());
+                        return false;
+                    default:
+                        throw new RuntimeException("Unexpected status from /aws2-lambda/function/create "
+                                + response.statusCode() + " " + response.body().asString());
+                    }
+                });
+
```

Comment:
The idea is to keep function creation/deletion at this level. And push other functions/alias tests in sub-methods.